src/core/reader/types/element.rs

```diff
@@ -0,0 +1,165 @@
+use core::fmt::Debug;
+
+use alloc::vec::Vec;
+
+use crate::{
+    core::reader::span::Span, read_constant_expression::read_constant_instructions, Error, Result,
+};
+
+use super::RefType;
+
+#[derive(Clone)]
+pub struct ElemType {
+    pub init: ElemItems,
+    pub mode: ElemMode,
+}
+
+impl Debug for ElemType {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "ElemType {{\n\tinit: {:?},\n\tmode: {:?},\n\t#ty: {}\n}}",
+            self.init,
+            self.mode,
+            self.init.ty()
+        )
+    }
+}
+
+impl ElemType {
+    pub fn ty(&self) -> RefType {
+        self.init.ty()
+    }
+    pub fn to_ref_type(&self) -> RefType {
+        match self.init {
+            ElemItems::Exprs(rref, _) => rref,
+            ElemItems::RefFuncs(_) => RefType::FuncRef,
+        }
+    }
+
+    pub fn read_from_wasm(
```

Comment:
We couldn't just blindly implement `WasmReadable` for ElemType, as we need to pass multiple arguments to the reading function